### ISSUE TITLE
When creating a new Rails 8.0 app with `rails new`, what is the best practice?

### ISSUE THREAD

**Issue by jbwl** [inertiajs/inertia-rails#159]
Hello, first of all thank you for this gem.

I think it would be helpful to include the installation instructions for a brand new Rails app (8.0 or 7.2), especially what options / flags to use for the `rails new` command in order to be able to use inertia-rails with Kamal 2, Propshaft, esbuild and Tailwind.

Specifically, I am wondering if I need to specify Tailwind with `rails new` because the inertia-rails generator also installs Tailwind.
Then, will inertia-js work with esbuild? And Propshaft (the new default)? And in turn, the Dockerfile that is created along the Rails app, will I be able to deploy with Kamal?

The relevant `rails new` flags in question are:

```
# Choose JavaScript approach
# Default: importmap
# Possible values: importmap, bun, webpack, esbuild, rollup
-j, --js,      [--javascript=JAVASCRIPT] 

# or do I have to skip Javascript completely?
-J  # Skip JavaScript files

# Choose CSS processor. Check https://github.com/rails/cssbundling-rails for more options
# Possible values: tailwind, bootstrap, bulma, postcss, sass
-c,            [--css=CSS] 

# Skip Kamal setup (because it might not be compatible with inertia-rails anyway?)
[--skip-kamal]                                                

# Asset pipeline - do I need to skip it?
-A   # Indicates when to generate skip asset pipeline
```

TLDR; what options do I have to choose when doing `rails new` with Rails 8.0 when I want to use inertiajs, but also be able to use Rails 8 goodies like Kamal and Propshaft and esbuild?

Thank you!

**Comment by csalmeida** [comment]
I am not sure if compatibility with Kamal 2 is still an issue or not but I have deployed a Rails 8 with Inertia 2 and Svelte 5 recently and I got the following error to do with the assets. I have found a solution so will post here in case someone else hits it in the future.

```bash
#15 ERROR: process "/bin/sh -c SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile" did not complete successfully: exit code: 1
------
 > [build 6/6] RUN SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile:
1.080   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
1.080
1.080 No such file or directory - npx
1.080
1.080
1.080 Caused by:
1.081 Errno::ENOENT: No such file or directory - npx (Errno::ENOENT)
1.081
1.081 Tasks: TOP => vite:build_all
1.081 (See full trace by running task with --trace)
------
Dockerfile:49
--------------------
  47 |
  48 |     # Precompiling assets for production without requiring secret RAILS_MASTER_KEY
  49 | >>> RUN SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile
  50 |
  51 |
--------------------
ERROR: failed to solve: process "/bin/sh -c SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/kamal-local-docker-container/kamal-local-docker-container0/wrg6cishp8j4gcs5vl1oqpop1
```

To fix itn I had to add Node and install the modules so that Vite is present:

```dockerfile
curl -fsSL https://deb.nodesource.com/setup_20.x | bash - && \
    apt-get install -y nodejs && \
```

```dockerfile
# Install Node.js dependencies
RUN npm ci
```

Here's the full file:

```dockerfile
# syntax=docker/dockerfile:1
# check=error=true
# This Dockerfile is designed for production, not development. Use with Kamal or build'n'run by hand:
# docker build -t maset .
# docker run -d -p 80:80 -e RAILS_MASTER_KEY=<value from config/master.key> --name maset maset
# For a containerized dev environment, see Dev Containers: https://guides.rubyonrails.org/getting_started_with_devcontainer.html
# Make sure RUBY_VERSION matches the Ruby version in .ruby-version
ARG RUBY_VERSION=3.2.2
FROM docker.io/library/ruby:$RUBY_VERSION-slim AS base

# Rails app lives here
WORKDIR /rails

# Install base packages
RUN apt-get update -qq && \
    apt-get install --no-install-recommends -y curl libjemalloc2 libvips sqlite3 && \
    rm -rf /var/lib/apt/lists /var/cache/apt/archives
    
# Set production environment
ENV RAILS_ENV="production" \
    BUNDLE_DEPLOYMENT="1" \
    BUNDLE_PATH="/usr/local/bundle" \
    BUNDLE_WITHOUT="development"
    
# Throw-away build stage to reduce size of final image
FROM base AS build

# Install packages needed to build gems
RUN apt-get update -qq && \
    apt-get install --no-install-recommends -y build-essential git libyaml-dev pkg-config && \
    curl -fsSL https://deb.nodesource.com/setup_20.x | bash - && \
    apt-get install -y nodejs && \
    rm -rf /var/lib/apt/lists /var/cache/apt/archives
    
# Install application gems
COPY Gemfile Gemfile.lock ./
RUN bundle install && \
    rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git && \
    bundle exec bootsnap precompile --gemfile
    
# Copy application code
COPY . .

# Install Node.js dependencies
RUN npm ci

# Precompile bootsnap code for faster boot times
RUN bundle exec bootsnap precompile app/ lib/

# Precompiling assets for production without requiring secret RAILS_MASTER_KEY
RUN SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile

# Final stage for app image
FROM base

# Copy built artifacts: gems, application
COPY --from=build "${BUNDLE_PATH}" "${BUNDLE_PATH}"
COPY --from=build /rails /rails

# Run and own only the runtime files as a non-root user for security
RUN groupadd --system --gid 1000 rails && \
    useradd rails --uid 1000 --gid 1000 --create-home --shell /bin/bash && \
    chown -R rails:rails db log storage tmp
USER 1000:1000

# Entrypoint prepares the database.
ENTRYPOINT ["/rails/bin/docker-entrypoint"]

# Start server via Thruster by default, this can be overwritten at runtime
EXPOSE 80
CMD ["./bin/thrust", "./bin/rails", "server"]
```
